ios/testmanagerd/testlistener.go

```diff
@@ -0,0 +1,210 @@
+package testmanagerd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/danielpaulus/go-ios/ios/nskeyedarchiver"
+	log "github.com/sirupsen/logrus"
+)
+
+// Realtime test callbacks and test results
+type TestListener struct {
+	testFinishedChannel chan struct{}
+	err                 error
+	logWriter           io.Writer
+	debugLogWriter      io.Writer
+	testSuite           *TestSuite
+}
+
+type TestSuite struct {
+	Name          string
+	StartDate     time.Time
+	EndDate       time.Time
+	TestDuration  time.Duration
+	TotalDuration time.Duration
+	TestCases     []TestCase
+}
+
+type TestCase struct {
+	ClassName  string
+	MethodName string
+	Status     TestStatus
+	Duration   time.Duration
+	// TODO : add attachments from xcActivityRecord
+}
+
+const (
+	StatusFailed          = "failed"           // Defined by Apple
+	StatusPassed          = "passed"           // Defined by Apple
+	StatusExpectedFailure = "expected failure" // Defined by Apple
+	StatusStalled         = "stalled"          // Defined by us
+
+	// Test suite counter constants
+	UnknownCount uint64 = 0
+)
+
+type TestStatus struct {
+	Status string
+	Err    TestError
+}
+
+type TestError struct {
+	Message string
+	File    string
+	Line    uint64
+}
+
+func NewTestListener(logWriter io.Writer, debugLogWriter io.Writer) *TestListener {
+	return &TestListener{
+		testFinishedChannel: make(chan struct{}),
+		logWriter:           logWriter,
+		debugLogWriter:      debugLogWriter,
+	}
+}
+
+func (t *TestListener) didFinishExecutingTestPlan() {
+	close(t.testFinishedChannel)
+}
+
+func (t *TestListener) initializationForUITestingDidFailWithError(err nskeyedarchiver.NSError) {
+	t.err = err
+	close(t.testFinishedChannel)
+}
+
+func (t *TestListener) didFailToBootstrapWithError(err nskeyedarchiver.NSError) {
+	t.err = err
+	close(t.testFinishedChannel)
+}
+
+func (t *TestListener) testCaseStalled(testClass string, method string, file string, line uint64) {
+	log.Debug("TODO ?")
+
+	testCase := t.testSuite.findTestCase(testClass, method)
+	if testCase != nil {
+		testCase.Status = TestStatus{
+			Status: StatusStalled,
+			Err: TestError{
+				Message: "Test case stalled",
+				File:    file,
+				Line:    line,
+			},
+		}
+	}
+}
+
+func (t *TestListener) testCaseFinished(testClass string, testMethod string, xcActivityRecord nskeyedarchiver.XCActivityRecord) {
+	log.Debug("TODO ?") // Try screenshots
```

Comment:
Here we could have even a `Warn` message that we don't know how to handle it yet, otherwise it's hard to detect those cases

Reply:
Added.